server/src/main/java/io/crate/metadata/cluster/RenameTableClusterStateExecutor.java

```diff
@@ -67,7 +69,18 @@ public ClusterState execute(ClusterState currentState, RenameTableRequest reques
         Metadata currentMetadata = currentState.metadata();
         Metadata.Builder newMetadata = Metadata.builder(currentMetadata);
         ViewsMetadata views = currentMetadata.custom(ViewsMetadata.TYPE);
-        if (views != null && views.contains(source)) {
+        boolean isView = views != null && views.contains(source);
+
+        boolean viewExists = views != null && views.contains(target);
+        boolean tableExists = currentMetadata.hasIndex(target.indexNameOrAlias());
```

Comment:
maybe worth testing all 3 scenarios locally but if it works (or if some fix is needed and covers all 3) to add only 1 test

Reply:
I think it's better to cover all cases, to avoid future issues if the code is changed, this kind of stuff can lead to scary situations with unreadable cluster state and so on, so I'm adding all combinations.